lesson-05

pmobile83 opened the pull request:
3. подумать какие объекты в ТЗ вы считаете критичными для решения задачи
4. подумать какими бы классама вы описали эти объекты
5. подумать какими бы атрибутами вы наделили эти классы
6. подумать в какие бы пакете вы разместили эти классы
7. подумать какие бы методами описали взаимодействия классов
8. сделать отдельную ветку от develop
9. в мавен проекте сделать консольное приложение  и описать в нем артефакты из пунктов 3-7
10. сделать пул реквест в develop

pmobile83 committed:
3. подумать какие объекты в ТЗ вы считаете критичными для решения задачи
4. подумать какими бы классама вы описали эти объекты
5. подумать какими бы атрибутами вы наделили эти классы
6. подумать в какие бы пакете вы разместили эти классы
7. подумать какие бы методами описали взаимодействия классов
8. сделать отдельную ветку от develop
9. в мавен проекте сделать консольное приложение  и описать в нем артефакты из пунктов 3-7
10. сделать пул реквест в develop

RodinDenis commented on pom.xml:
```diff
@@ -4,13 +4,47 @@
+    <artifactId>Project-ATM</artifactId>
+    <version>1.0.1</version>
+
+    <build>
```
обычно эту секцию размещают ближе к  концу файла

RodinDenis commented on pom.xml:
```diff
@@ -4,13 +4,47 @@
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>RELEASE</version>
```
лучше указывать конкретную версию

RodinDenis commented on src/main/java/ru/sber/javaschool/atmfunction/GetBalance.java:
```diff
@@ -0,0 +1,17 @@
+public class GetBalance {
+
+    public static double getBalance() {
+        double balance = 0.0;
```
для сумм используем только BigDecimal

RodinDenis commented on src/main/java/ru/sber/javaschool/verification/VerificationType.java:
```diff
@@ -0,0 +1,9 @@
+
+/* Верификация пользователя на банкомате: Card, NFC, FaceId, Quar */
+public enum VerificationType {
+    Card,
```
нарушен нейминг. они должны быть как константы быть в верхнем регистре

RodinDenis commented on src/main/java/ru/sber/javaschool/users/SelfClient.java:
```diff
@@ -0,0 +1,13 @@
+class SelfClient extends User {
+
+    public SelfClient(String cardnumber, int pinCode) {
+        super(cardnumber = "4444555566660000", pinCode = 9999);
```
не совсем понятно какой эффект тут хочется получить. есть подозрение что результат будет отличаться от ожиданий автора. Рекомендую исполнить этот этот кусок и посмотреть на результат. Если будут вопросы, то обсудим голосом

pmobile83 replied:
Согласен, был не прав )
Отправлен commit#2 с исправлениями.

pmobile83 committed:
Исправление полученных замечаний в рамках lesson-5:
 - перенос секции <build> в конец pom.xml
 - указание конкретной версии зависимости в pom.xml
 - переход к использованию BigDecimal в суммах
 - исправление нэйминга в ENUM

 Выполнение первой части заданий в рамках lesson-10 (создание модели).

pmobile83 committed:
Добавлен класс Balance

RodinDenis commented on src/main/java/com/github/baklashovanton/atm/Console.java:
```diff
@@ -0,0 +1,35 @@
+        account2.addCard("2222 2222 3333 4444");
+        account2.addCard("3333 3333 4444 5555");
+
+        client1.addAccount(account1);
```
это не проблема, но я бы сначала создал счета и добавил их в карты до того как помещать карты в процессинг. Мне кажется так проще

RodinDenis commented on src/main/java/com/github/baklashovanton/atm/atm/Function.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.math.BigDecimal;
+
+public class Function {
```
я бы избегал таких названий. Лучше выбрать ближе к предметной области

RodinDenis commented on src/main/java/com/github/baklashovanton/atm/atm/Function.java:
```diff
@@ -0,0 +1,13 @@
+import java.math.BigDecimal;
+
+public class Function {
+    private static BigDecimal balance;
```
поле не используется

RodinDenis commented on src/main/java/com/github/baklashovanton/atm/client/Client.java:
```diff
@@ -0,0 +1,23 @@
+
+import lombok.Data;
+
+@Data
```
есть поля содержащие коллекции. Отдавать их нельзя так как это нарущит инкапсуляцию. Лучше либо делать бизнес методы оперирующие коллекцией либо отдавать стрим. Раньше делали неизменяемой коллекцию до появления стримов но это хуже так как клиент завязывается на конкретную коллекцию

pmobile83 committed:
Финальный проект реализованный в рамках курса "Школа Java Developer".
